components/squad/wikis/starcraft2/squad_custom.lua

```diff
@@ -54,6 +54,7 @@ function CustomSquad.run(frame)
 	while args['p' .. index] or args[index] do
 
 		local player = Json.parseIfString(args['p' .. index] or args[index])
+		player.id = player.id or player[1] or player.name
```

Comment:
Could you explain this part? The choice to require `|id` was very deliberate

Reply:
because for some (very old) team pages we do not have ids, just names
and for the other ones it is (in the old templates) entered as the first unnamed param
so to be able to easily convert them the `or pülayer[1]` is wanted
the `or player.name` for the old legacy cases

Reply:
If a player does not have an id I don't think `:id()` should be called?

Reply:
i removed the `or player.name` for now, will open an additional PR for that later today

Reply:
okay is only on 2 pages ...

Reply:
Alright open for solutions in a new PR :)

Reply:
Oh wait, I thought you removed the whole thing. I still don't see why we're including the `player[1]` part?

Reply:
because until now we used the first unnamed arg instead of `|id=` for the id
it basically is just an alias so that conversion is easily doable via bot
alternative would be to overwrite the `Person` template on sc2 (and probably sc) with setting `|id={{#if:{{{id|}}}|{{{id}}}|{{{1|}}}}}`

Reply:
Are we doing conversion per bot? We're not doing that on RL.

Reply:
i don't want to convert ~500 pages manually
so yeah i wanted to do it via bot
```python
python pwb.py replace -regex -transcludes:"Squad player row" "former=true" "status=former" "inactive=true" "status=inactive" "\{\{Squad player list start(.*?)\}\}" "{{Squad \1" "\{\{Squad player row" "|{{Person" "\{\{Squad player list end\}\}" "}}"
```

Reply:
If we made this change temporarily, would it be possible to bot the change, and then do another run where we add `|id=`?

Reply:
if you come up with a regex that works we can do it, but i don't see an easy one
the order of the args within the template calls are different from time to time so we can not even say that the id is always after the race or such heuristics


(that all is ignoring the params that will just get ignored by the new one)

Reply:
Can we not just grab the first param without a param name? First attempt: `\{\{Squad player row|\|([^=]*)\|`

Else, I think we would first restructure the order, so e.g. `|race` always after `|name`. Do that enough and we should reach consistency. Then we can pretty easily get a regex to add `|id=`.

Reply:
will test around a bit

Reply:
yeah found a regex that should work on most pages (having to use dotall due to the additional old stuff we have in them)
so was a bit more complicated so it doesn't run outside of the squad player row calls
i adjusted the old module so it allows id as input
will start the first bot run (adding the id stuff) now, and after that one (and the merge) i will start the switch run

so now the PR is just the change in the requires

Reply:
Great!